pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -353,6 +394,53 @@ func logbackFromFile(t string, data interface{}) (string, error) {
 	return output.String(), nil
 }
 
+// Adds a component list object in state in progress. This will cause UpdateStatus to update the
+// ready type to state "false".
+func addUpgradeComponentNotReady(master *alpha1.CDAPMaster) {
+	os := status.ObjectStatus{}
+	os.Name = "upgrade"
+	os.Status = status.StatusInProgress
+	master.Status.ComponentMeta.ComponentList.Objects = append(master.Status.ComponentMeta.ComponentList.Objects, os)
+}
+
+func launchJob(master *alpha1.CDAPMaster, rsrclabels map[string]string) ([]reconciler.Object, error) {
```

Comment:
Instead of calling it `launchJob`, better call it as `prepareNewPreUpgradeJobResources`